### ISSUE TITLE
model pretraining throws error 

### ISSUE THREAD

**Issue by JonathanBhimani-Burrows** [manujosephv/pytorch_tabular#369]
**Describe the bug**
bug occurs 

Trying to set up pretraining for a model following the API here https://pytorch-tabular.readthedocs.io/en/latest/tabular_model/#training-api-self-supervised-learning

**To Reproduce**
1. set up code 
**data_config = DataConfig(
        target=['Age_Group_Target'], #target should always be a list.
        continuous_cols=self.num_cols,
        categorical_cols=self.cat_cols,
        num_workers=8
    )

    trainer_config = TrainerConfig(
        # auto_lr_find=True, # Runs the LRFinder to automatically derive a learning rate
        batch_size=self.batch_size,
        max_epochs=self.max_epochs,
        early_stopping=None,
        # early_stopping="valid_loss", # Monitor valid_loss for early stopping
        # early_stopping_mode = "min", # Set the mode as min because for val_loss, lower is better
        # early_stopping_patience=self.early_stopping_patience, 
        checkpoints="valid_loss", # Save best checkpoint monitoring val_loss
        load_best=True, # After training, load the best checkpoint
        accelerator=self.device,
        deterministic=True
    )

    optimizer_config = OptimizerConfig(optimizer="AdamW")

    head_config = LinearHeadConfig(
          layers=layers, # No additional layer in head, just a mapping layer to output_dim
          dropout=dropout_prob,
          initialization="xavier",
          # activation="ReLU"
      ).__dict__ # Convert to dict to pass to the model config (OmegaConf doesn't accept objects)

    print(head_config)
    model_config = CategoryEmbeddingModelConfig(
          task="ssl",
          layers="",  # Number of nodes in each layer
          activation="ReLU", # Activation between each layers
          head = "LinearHead", #Linear Head
          head_config = head_config, # Linear Head Config
          dropout=0.1,
          metrics = ['f1_score'],
          metrics_params = [ {'num_classes': 2, 'average': 'macro'}],
          metrics_prob_input = [True]
      )

    tabular_model = TabularModel(
        data_config=data_config,
        model_config=model_config,
        optimizer_config=optimizer_config,
        trainer_config=trainer_config,
        #  experiment_config=experiment_config
    )

    tabular_model.pretrain(train=train, validation=test, max_epochs=1)**

2. Run 

**Expected behavior**
Pretraining happens 

**Actual behavior**
_NotImplementedError: ssl is not a valid task. Should be one of ['regression', 'classification', 'backbone']_
Ok so I switched to 'backbone', and get the following error 
_AssertionError: `pretrain` is not valid for backbone task. Please use `fit` instead._

Either there's a bug or my config is wrong - any help is appreciated! Thanks :) 
If you can help, I can add a tutorial on how to use pretraining for others in the future!

**Screenshots**

**Desktop (please complete the following information):**

- OS:Windows 10
- Browser Chrome (Google Colab)
- Version pytorch tabular 
- pytorch tabular release 1.0.2 OR installing from most recent git commit (slightly different errors)

**Smartphone (please complete the following information):**

- Device: \[e.g. iPhone6\]
- OS: \[e.g. iOS8.1\]
- Browser \[e.g. stock browser, safari\]
- Version \[e.g. 22\]

**Additional context**

**Comment by JonathanBhimani-Burrows** [comment]
It seems like the setup for SSL is specific to a few model architectures only, my mistake 
For anyone else, the tutorial is here 
https://pytorch-tabular.readthedocs.io/en/latest/ssl_models/
